### ISSUE TITLE
Cannot detect workspace root

### ISSUE THREAD

**Issue by Foddy** [nuxt/framework#8014]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.17.1`
- Nuxt Version:     `3.0.0-rc.9`
- Nitro Version:    `0.5.4`
- Package Manager:  `unknown`
- Builder:          `vite`
- User Config:      `globalName`, `builder`, `app`, `build`, `modules`, `postcss`, `css`, `components`, `vite`, `nitro`, `hooks`
- Runtime Modules:  `@vueuse/nuxt@9.3.0`, `nuxt-jsonld@2.0.6`
- Build Modules:    `-`

### Reproduction

Install the latest version and run `nuxi generate`

### Describe the bug

Since the RC version 9 or higher I always get this error when trying to generate a static version of my frontend. Downgrading to RC8 (or below) works as expected.

### Logs

```shell
$ cd /www/frontend
$ nuxi generate
Nuxi 3.0.0-rc.9                                                                                                                                                         10:51:27
Nuxt 3.0.0-rc.9 with Nitro 0.5.4                                                                                                                                        10:51:27

 ERROR  Cannot detect workspace root from /www/frontend                                                                                    10:51:28

  at findWorkspaceDir (node_modules/pkg-types/dist/index.mjs:117:9)
  at async loadOptions (node_modules/nitropack/dist/shared/nitro.9c828268.mjs:2271:26)
  at async createNitro (node_modules/nitropack/dist/shared/nitro.9c828268.mjs:2347:19)
  at async initNitro (node_modules/nuxt/dist/index.mjs:1373:17)
  at async initNuxt (node_modules/nuxt/dist/index.mjs:1559:3)
  at async loadNuxt (node_modules/nuxt/dist/index.mjs:1581:5)
  at async loadNuxt (node_modules/@nuxt/kit/dist/index.mjs:732:19)
  at async Object.invoke (node_modules/nuxi/dist/chunks/build.mjs:33:18)
  at async Object.invoke (node_modules/nuxi/dist/chunks/generate.mjs:31:5)
  at async _main (node_modules/nuxi/dist/cli.mjs:50:20)
```

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by Foddy** [comment]
Okay, I feel like it has something to do with my processor or OS. I can't reproduce this error on Stackblitz or my AMD based linux server.

The error occurs only on macOS 12.6 (21G115) with a M1 Max processor running node v16.17.1 installed with Homebrew and also on a M1 iMac with the same OS version :( …

I wish I could give you somehow private access to the affected machine so you can debug?

**Comment by danielroe** [comment]
See if you can reproduce in a minimal git repo. I can test it in a MacOS environment.

**Comment by hutchic** [comment]
reproduced [here](https://github.com/hutchic/nuxt-issue-8014)

```
git clone https://github.com/hutchic/nuxt-issue-8014
cd nuxt-issue-8014
docker build -t temp .

... SNIP ...

#6 49.06 $ nuxt prepare
#6 49.11 Nuxi 3.0.0-rc.11
#6 49.70 
#6 49.70  ERROR  Cannot detect workspace root from /
#6 49.70 
#6 49.70   at findWorkspaceDir (/node_modules/pkg-types/dist/index.mjs:117:9)
#6 49.70   at async loadOptions (/node_modules/nitropack/dist/shared/nitro.9c828268.mjs:2271:26)
#6 49.70   at async createNitro (/node_modules/nitropack/dist/shared/nitro.9c828268.mjs:2347:19)
#6 49.70   at async initNitro (/node_modules/nuxt/dist/index.mjs:1402:17)
#6 49.70   at async initNuxt (/node_modules/nuxt/dist/index.mjs:1612:3)
#6 49.70   at async loadNuxt (/node_modules/nuxt/dist/index.mjs:1635:5)
#6 49.70   at async loadNuxt (/node_modules/@nuxt/kit/dist/index.mjs:732:19)
#6 49.70   at async Object.invoke (/node_modules/nuxi/dist/chunks/prepare.mjs:30:18)
#6 49.70   at async _main (/node_modules/nuxi/dist/cli.mjs:50:20)
#6 49.70 
#6 49.71 error Command failed with exit code 1.
#6 49.71 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
------
executor failed running [/bin/sh -c yarn install]: exit code: 1
```

**Comment by andresmeidla** [comment]
TLDR: a simple `git init` fixed it for me 

in my case the initial `npm install` failed (project created with `npx nuxi init nuxt-app`)

[pkg-types](https://github.com/unjs/pkg-types/blob/main/src/index.ts#L65) tries to find the project folder using either:
* git/config
* lockifiles ("yarn.lock", "package-lock.json", "pnpm-lock.yaml", "npm-shrinkwrap.json")
* or tries to find the path via the supplied parameters
I didn't have git repo initialized nor any lockfiles, so the 3rd option should've resolved the path, but it didn't.
`findFile("/tmp/nuxt-issue-8014", {  startsWith: "/tmp/nuxt-issue-8014" });` will throw an error for some reason

**Comment by hutchic** [comment]
> copying yarn.lock inside dockerfile will probably solve the issue

if you mean something along the lines of [this](https://github.com/hutchic/nuxt-issue-8014/commit/0797d437583f890c1f4f76dd4e0655761b13fec5) that didn't [work](https://github.com/hutchic/nuxt-issue-8014/actions/runs/3214797393/jobs/5255436608)

**Comment by andresmeidla** [comment]
> > copying yarn.lock inside dockerfile will probably solve the issue
> 
> if you mean something along the lines of [this](https://github.com/hutchic/nuxt-issue-8014/commit/0797d437583f890c1f4f76dd4e0655761b13fec5) that didn't [work](https://github.com/hutchic/nuxt-issue-8014/actions/runs/3214797393/jobs/5255436608)

try with workdir other than root, for ex /app: ```
FROM node:16
WORKDIR /app
COPY ["yarn.lock", "package.json", "package-lock.json*", "./"]
RUN ls -la
RUN yarn install
```
built successfully for me at least

**Comment by hutchic** [comment]
can confirm that workaround [works](https://github.com/hutchic/nuxt-issue-8014/pull/2). IMO still a bug that should be resolved or documented for true resolution of this issue

**Comment by danielroe** [comment]
This should be resolved in the [edge channel](https://v3.nuxtjs.org/guide/going-further/edge-channel/), or in the next RC.

Let me know if not and I'll reopen.
